lib/filewatch/watch.rb

```diff
@@ -43,7 +43,8 @@ def subscribe(observer, sincedb_collection)
       reset_quit
       until quit?
         iterate_on_state
-        break if quit?
+        # Don't discover new files when files to read are known at the beginning
+        break if quit? || @settings.exit_after_read
```

Comment:
I think here there is a bug. Supposing we have `max_open_files` to 10 and the `@watched_files_collection` contains 100 elements, with `mode => read` and `exit_after_read => true` I expect that the pipeline process all the 100 files and exit. After a first loop on the first 10s we have `quit = false` or `exit_after_read = true` so the loop exit, while still 90 files are to be read.

Reply:
You are right. I added unit test to check this scenario and it's failing. I need to rethink how to change `subscribe` method to work when not all files are opened at the start.

Reply:
Hi @BigYellowHammer do you want some help or support in this change?

Reply:
hey @andsel help is always welcome :) Recently I didnt have time to take a look on the fix. I just created new unit test to confirm the issue, want me to push it?

Reply:
yes please, so we can start from there

Reply:
Hi @BigYellowHammer I fixed the case, not the loop continue to process files contained into `@watched_files_collection` until the collection is empty. The collection is filled on plugin startup and never refilled, note that `@watched_files_collection` in class `FileWatch::Watch` contains the full list of files found when the plugin started, so avoid additions done after the start.

Reply:
are we good to merge it?

Reply:
I would test it locally before, squash commits and then I'll merge, tomorrow I think if I don't find any bug